x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/ClusterStateRoleMapper.java

```diff
@@ -81,7 +83,28 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
     }
 
-    private Set<ExpressionRoleMapping> getMappings() {
+    public boolean hasMapping(String name) {
```

Comment:
These can definitely be cleaned up not to duplicate so much code but I ran out of time

Reply:
Did some minor clean up. I think it's nice that `hasMapping` returns early.